### PR TITLE
Issues 19 test repo gemlock

### LINKED ISSUE #19: Error scanning testRepo Gemfile.lock with latest Fedora (34)
Using the Dockerfile to build a test environment for deplist and scanning the example testRepo results in an error scanning the Gemfile.lock present there. The error is:

`strptime-0.2.4 requires ruby version ~> 2.0, which is incompatible with the current version, ruby 3.0.4p208`

The full list of commands to get there:

```
$ git log -n 1
commit 5dd6e228d4e7425d87ea2a2bbb56da80406b3696

$ podman build -t deplist .

$ podman run -v /home/jshepher/projects/deplist/test/testRepo:/testRepo:Z deplist -debug /testRepo
time="2022-07-15T00:23:13Z" level=debug msg="Checking /testRepo"
time="2022-07-15T00:23:13Z" level=debug msg="GetRubyDeps /testRepo/Gemfile.lock"
time="2022-07-15T00:23:17Z" level=debug msg="Fetching gem metadata from https://rubygems.org/..........\nstrptime-0.2.4 requires ruby version ~> 2.0, which is incompatible with the\ncurrent version, ruby 3.0.4p208\n"
exit status 5
```

### PR DESCRIPTION
**Pull request by jasinner**
Fixes https://github.com/RedHatProductSecurity/deplist/issues/19

### CODE REVIEW

**Comment by jasinner** [comment]
@sfowl I fixed the tests here. Let me know if there are any problems with merging this.

File: Dockerfile
Comment (sfowl) [inline Dockerfile:12]: Why were ruby-devel and gcc-c-++ necessary? We're not currently including them in deptopia builds

File: .github/workflows/go.yml
Comment (sfowl) [inline .github/workflows/go.yml]: Is this related to the ruby problem? I think there's a bigger gap now between the go version in our dockerfiles and in the ci (v1.16)

File: deplist_test.go
Comment (sfowl) [inline deplist_test.go:150]: nitpick (for previous committer): shame these weren't sorted so that it was easier to see the changes

File: Dockerfile
Comment (jasinner) [reply Dockerfile:12]: I found that that bundler command used by deplist didn't work in the container without these dependencies. I think they are required to compile native extensions in 'unf' gem.

File: .github/workflows/go.yml
Comment (jasinner) [reply .github/workflows/go.yml]: I'm not sure what you mean by ci, isn't this file the one that dictates what used by ci? I upgraded to this version to match the one found in latest Fedora 'golang-bin' rpm.

File: deplist_test.go
Comment (jasinner) [reply deplist_test.go:150]: yeah, I basically replaced the ruby 2 project with a ruby 3 one. They are both based on fluentd, but this one is quite a bit different.

File: .github/workflows/go.yml
Comment (sfowl) [reply .github/workflows/go.yml]: I mean that this file, for our CI, now specifies go 1.18, whereas the Dockerfile still specifies go 1.16.

Comment (jasinner) [reply .github/workflows/go.yml]: I also updated the Dockerfile to use the default golang-bin package from Fedora (currently 1.18).

File: Dockerfile
Comment (jasinner) [reply Dockerfile:12]: I seems it's actually the unf-ext gem.

File: Dockerfile
Comment (sfowl) [inline Dockerfile]: The default golang-bin version in fedora 34 is 1.16. I don't think this changes much aside from making it less obvious which go version is installed, based on the Dockerfile.

File: .github/workflows/go.yml
Comment (sfowl) [reply .github/workflows/go.yml]: @jasinner sorry, I'm not try to hold this PR up, overall this is seems to be a helpful change. I'll try to clarify my intention in the original comment:

a) is updating the Go version a required part of this change, to fix the ruby issue?

b) If it is, can we try to make the go version consistent in all places referenced in this repo?

c) If it is not, would it be easier to revert to Go version changes, and just fix the ruby problem?

Comment (jasinner) [reply .github/workflows/go.yml]: @sfowl  I reverted the go version change commit as it's not required to fix the issues described in issue/19. 

The reason I added it was because the README says to install golang-bin, which if you do that with the latest Fedora image you got 1.18. Therefore went around updating the go version in other places to make that consistent. However now I've updated the README to instruct users to install golang-bin-1.16 instead.

File: README.md
Comment (sfowl) [inline README.md]: This won't work on Fedora versions without go 1.16 (e.g. Fedora 36).

File: .github/workflows/go.yml
Comment (sfowl) [reply .github/workflows/go.yml]: Ok thanks for the explanation!

It seems deplist is not very consistent in it's usage of Go versions at present. I see four different places:
* ci: 1.17
* go.mod: 1.17
* dockerfile: 1.16
* readme: undefined (though perhaps implicitly following go.mod?)

I think we should try to bring these all up to the same version at once, but should probably be done separately to this PR, if it's not needed. If you can undo the readme change, I'm happy to merge.

File: README.md
Comment (sfowl) [inline README.md]:
```suggestion
$ dnf install golang-bin yarnpkg maven rubygem-bundler ruby-devel gcc gcc-c++ npm
```

**Review by sfowl — APPROVED** [review]

**Pull request merged**
